sympy/core/tests/test_cache.py

```diff
@@ -74,3 +75,13 @@ def prop(self):
     assert a.calls == 1
     b = A(None)
     assert b.prop == None
+
+
+def test_lazy_function():
+    module_name='xmlrpc.client'
+    function_name = 'gzip_decode'
+    lazy = lazy_function(module_name, function_name)
+    assert module_name not in sys.modules
+    assert lazy(b'') == b''
+    assert module_name in sys.modules
```

Comment:
I guess this test would fail if run twice in the same process. Maybe it's enough just to check that the doc matches and the function gets called.

Reply:
@oscarbenjamin It would fail indeed. Not relevant for the command line testing, but if someone calls `sympy.test()` twice in a single python session it will fail. I can remove the test, or replace it with something like
```
from sympy.core.cache import lazy_function
import sys
from xmlrpc.client import gzip_decode

module_name='xmlrpc.client'
function_name = 'gzip_decode'
cnt=sys.getrefcount(gzip_decode)
print(f'before lazy_function: sys.getrefcount(gzip_decode) {sys.getrefcount(gzip_decode)}')
lazy = lazy_function(module_name, function_name)
cnt1=sys.getrefcount(gzip_decode)
print(f'after lazy_function: sys.getrefcount(gzip_decode) {sys.getrefcount(gzip_decode)}')
assert cnt==cnt1
```
(this will not fail it called twice, but it could fail if some multi-threading is going on)

Let me know what your preference is.

Reply:
One possibility would be to run the test in a subprocess. Otherwise I don't think we need to test this that comprehensively. The real thing that we should be keeping track of is import time since minimising that is the goal here.

Reply:
We probably are going to want to have some tests for lazy importing in general, which will have to run in a subprocess. This is similar to the `bin/test_submodule_imports.py` and the `bin/test_optional_dependencies.py` tests, which import sympy in a subprocess to check for import side effects. In fact they could perhaps all be combined.

Reply:
@asmeurer I added the test to `bin/test_lazy_function.py`.